### ISSUE TITLE
Host individual DEP repos under dart-lang org

### ISSUE THREAD

**Issue by nex3** [dart-lang/dart_enhancement_proposals#15]
Currently the pattern for DEP repos is to host them under the proposer's personal GitHub accounts. This is fine when preparing the dep to send out, but once it's officially in the process it should be in the dart-lang organization. It's easy for Googlers to do this for their own deps, and for external DEPs a Googler can fork the original repo and add the user as an owner.

Partly this is for organizational neatness, but it's also a big help when configuring notification emails. I want all my Dart-related emails to go to my work account, but that's only configurable on a per-organization basis. Right now, I'm watching [the configured imports DEP](https://github.com/lrhn/dep-configured-imports) and I missed a bunch of discussion because the notifications went to the wrong place.

**Comment by munificent** [comment]
> It's easy for Googlers to do this for their own deps, and for external DEPs a Googler can fork the original repo and add the user as an owner.

Do we allow non-Googlers to be owners of repos under dart-lang? I didn't think we did.

> Right now, I'm watching the configured imports DEP and I missed a bunch of discussion because the notifications went to the wrong place.

Oh, that's super annoying. :(

**Comment by nex3** [comment]
> Do we allow non-Googlers to be owners of repos under dart-lang? I didn't think we did.

I'm just saying give them commit access.

**Comment by munificent** [comment]
I can bring this up with the committee, but my hunch is it won't work. I fear having a outside-authored-DEP (which is a use case I really care about) living under dart-lang would carry an implication that it has a certain "stamp of approval" or official status that it doesn't have (yet).

**Comment by mezoni** [comment]
>> DEP (which is a use case I really care about) living under dart-lang would carry an implication that it has a certain "stamp of approval" or official status that it doesn't have (yet).

Your conclusion are wrong.
The "approval" is just a state (or stage).
Look at the Python PEP:

1.
PEP numbers are assigned by the PEP editors, and once assigned are never changed

2.
Index by Category
num title owner
Meta-PEPs (PEPs about PEPs or Processes)
...items
Other Informational PEPs
...items
Accepted PEPs (accepted; may not be implemented yet)
...items
Open PEPs (under consideration)
...items
Finished PEPs (done, implemented in code repository)
...items
 Abandoned, Withdrawn, and Rejected PEPs
...items

Example of item

IW | 206 | Python Advanced Library | Kuchling

3.
Numerical Index

4.
Reserved PEP Numbers

5.
Key
    S - Standards Track PEP
    I - Informational PEP
    P - Process PEP

    A - Accepted proposal
    R - Rejected proposal
    W - Withdrawn proposal
    D - Deferred proposal
    F - Final proposal
    A - Active proposal
    D - Draft proposal
    S - Superseded proposal

6.
Owners
    name                         email address
    ----                         -------------
    Aahz                         aahz at pythoncraft.com

7.
References

**Github is right place???***

**Comment by mezoni** [comment]
You even don't try to implement the generator of static documents (eg. markdown) that may show info in one place (even on github).
This is example of this generated document: https://www.python.org/dev/peps

This is a static (autogenerated) page.
Discussions (topic per proposal) can be performed in some mail-lists.
One mail-list is meta mail-list and another mail-list (topic per proposal) for proposals (topic created only once).
Refrence to the topic in mail-list can by in the proposal iindex document (described above).

**Comment by NDxTreme** [comment]
Perhaps a structural change to the process? Instead of each of us starting our own repository, we Fork this one. We ten create a new document that describes our proposal, and create a pull request. the pull request gives us a place to talk about the changes with the community, and the proposal can be added or dropped after it is accepted?

At least as a start of the idea.

Or, each of us create a branch, create our document, and only documents that are promoted to master are accepted as official. The originator would still need to create a pull request, and the only substance given would be the willingness to talk about it.

**Comment by munificent** [comment]
> We ten create a new document that describes our proposal, and create a pull request. the pull request gives us a place to talk about the changes with the community, and the proposal can be added or dropped after it is accepted?

That was my initial thought, but long-running pull requests are pretty terrible to manage. Having the proposal be its own repo makes it easy to have separate discussion threads about different parts of the proposal. It also makes it easier for people to collaborate on a proposal.

Forking this repo doesn't add any value as far as I can tell. It's not like each proposal really needs its own copy of stuff like the DEP committee meeting notes.

**Comment by nex3** [comment]
I believe it also wouldn't solve my problem, since issues on the fork and I think even comments on the PR will be under the aegis of the fork, not the root repo.

**Comment by NDxTreme** [comment]
Then, we could always have the discussion here, in the threads that announce the proposals.

**Comment by munificent** [comment]
> Then, we could always have the discussion here, in the threads that announce the proposals.

A single issue doesn't handle multiple unrelated threads of discussion very well.

**Comment by seaneagan** [comment]
I think it's fine if google wants to require its employees to file DEPs under the dart-lang organization.  
IMO non-googler DEP authors will want to host their DEPs under their own account.  This will motivate non-googler DEPs since one can more easily socialize their contribution, and allows the author to control access rights etc.  Once it is `Accepted` then it could be forked into a `tc52` or `dart-lang` organization.

**Comment by mezoni** [comment]
@seaneagan 

>> IMO non-googler DEP authors will want to host their DEPs under their own account. 

You want to say that non-googler DEP always are unofficial?

Please, read about the Python PEP.
Them not divided in two major parts:

- Guido van Rossum PEP
- non-Guido van Rossum PEP

Why need this discrimination?
All DEP should be official regardless of authorship.

>> This will motivate non-googler DEPs since one can more easily socialize their contribution

It they will be considered as the official DEP's, how one person (author) can manage (change) DEP personally if DEP are official?

Or DEP in Dart mostly personal?
I am talk only about the text of the DEP.

**Comment by mezoni** [comment]
@seaneagan 

>> This will motivate non-googler DEPs since one can more easily socialize their contribution

DEP is only textual (read-only) form of the proposal.

`More easily socialize their contribution` author can and should in appropriate forums.

Internet forum
http://en.wikipedia.org/wiki/Internet_forum
An Internet forum, or message board, is an `online discussion` site where people can hold `conversations in the form of posted messages`.

P.S.
At least in mail-list.

**Comment by NDxTreme** [comment]
@mezoni The point of moving these discussions to Github, at least as I understand it, was to provoke more communication from the community, since the already existing mailing list was essentially dead.

The Google Groups is essentially a forum/newsgroup/mailing list, and it was not receiving any response. This move has already provoked more discussion in its first week than the entire message board.

**Comment by mezoni** [comment]
@NDxTreme 
>> since the already existing mailing list was essentially dead.

Interesting point of view.
This is your personal opinion or this is a fact (known to all)?

>> This move has already provoked more discussion in its first week than the entire message board.

Only by a very few number of persons.
Only by a very few...
...and without any positive results.
What happens when these (a very few number of persons that expressed interest) will lose interest?

Github is not right place for discussions.

**Comment by munificent** [comment]
> Once it is Accepted then it could be forked into a tc52 or dart-lang organization.

Yes, this is a [documented step in the process](https://github.com/dart-lang/dart_enhancement_proposals/blob/master/Walkthrough.md#acceptance). :)

**Comment by mezoni** [comment]
First impressions are simply disgusting.
It's not interesting and highly qualified specialists will not discuss anything here with you.
Do not think that I talk about me.
No, I am an amateur.
I'm talking about those professionals who turned away by the Dart Team in a far in 2012 due to the uncompromising of the Dart Team.
I like Dart but I do not like some Dart developers because they are a burden and stubborn.

**Comment by munificent** [comment]
Closing this out. I realize the notifications aren't ideal, but I think the current setup is the simplest one for now.
